packages/graphql-validate-fixtures/src/index.ts

```diff
@@ -1,68 +1,181 @@
+  {cwd}: Options,
 ): Promise<Evaluation[]> {
-  let schema: GraphQLSchema;
+  const config = getGraphQLConfig(resolve(cwd));
 
-  try {
-    const schemaJSON = await readJSON(schemaPath, {encoding: 'utf8'});
-    schema = buildClientSchema(schemaJSON.data);
-  } catch (error) {
-    throw new Error(
-      `Error parsing '${schemaPath}':\n\n${error.message.replace(
-        /Syntax Error GraphQL \(.*?\) /,
-        '',
-      )}`,
-    );
+  const projectOperations = await Promise.all(
+    getGraphQLProjects(config).map(getOperationsForProject),
+  );
+
+  return runForEachFixture(fixturePaths, (fixture) =>
+    evaluateFixture(fixture, projectOperations),
+  );
+}
+
+interface FixtureOperation {
+  ast: AST;
+  error?: any;
+  operation: Operation;
+  operationName: string;
+}
+
+interface FixtureOperationError {
+  ast: AST;
+  error: any;
+  operation?: Operation;
+  operationName?: string;
+}
+
+function evaluateFixture(
+  fixture: Fixture,
+  projectOperations: GraphQLProjectOperations[],
+): Evaluation {
+  const fixtureOperations = projectOperations.map<
+    FixtureOperation | FixtureOperationError
+  >(({ast}) => {
+    try {
+      return {
+        ast,
+        ...getOperation(fixture, ast),
+      };
+    } catch (error) {
+      return {
+        ast,
+        error,
+        operation: undefined,
+        operationName: undefined,
+      };
+    }
+  });
+
+  const validFixtureOperations = fixtureOperations.filter(
+    ({operation, error}) => {
+      return operation != null && error == null;
+    },
+  ) as FixtureOperation[];
+
+  if (validFixtureOperations.length > 1) {
+    const [{operationName}] = validFixtureOperations;
+    const projectNames = projectOperations
+      .map(({config}) => config.resolveProjectName())
+      .join(', ');
+
+    return {
+      fixturePath: fixture.path,
+      validationErrors: [
+        new Error(
+          `Ambiguous operation name '${operationName}' in '${
+            fixture.path
+          }' for projects: ${projectNames}`,
+        ),
+      ],
+    };
   }
 
-  if (schemaOnly) {
-    return runForEachFixture(fixturePaths, (fixture) =>
-      validateFixtureAgainstSchema(fixture, schema),
-    );
+  if (validFixtureOperations.length === 0) {
+    let errors = fixtureOperations
+      .filter(({operation}) => operation != null)
+      .map(({error}) => error);
+
+    if (errors.length === 0) {
+      // we couldn't find an error in a valid fixture, fallback on all errors
+      errors = fixtureOperations.map(({error}) => error);
+    }
+
+    return {
+      fixturePath: fixture.path,
+      validationErrors: mergeMissingOperationErrors(errors),
+    };
+  }
+
+  const [{ast}] = validFixtureOperations;
+
+  return validateFixtureAgainstAST(fixture, ast);
+}
+
+function mergeMissingOperationErrors(errors: any[]) {
+  if (errors.length <= 1) {
+    return errors;
   }
 
-  const sources = await Promise.all(
+  const {otherErrors, operationNames} = errors.reduce<{
```

Comment:
IMO this is a weird way of doing it, operation on errors instead of projects.

Reply:
this is already a bit of a strange case. The `errors` have already been computed and we now just want to post-process the list. We don't have any typing information at this point so it's literally an array of zero or more things. We want to squeeze the errors list together and merge any `MissingOperationError`'s together, but leave other errors alone. The end result is an array of zero or more other errors, and zero or one Error representing the merge of `MissingOperationError`'s. I couldn't think of how to simplify this further, `reduce` was the best I could come up with keeping a mutating list of other errors and missing operation names.